typhon_test.go

```diff
@@ -0,0 +1,52 @@
+package typhon
+
+import (
+	"testing"
+	"time"
+
+	"github.com/b2aio/typhon/client"
+	"github.com/b2aio/typhon/example/proto/callhello"
+	"github.com/b2aio/typhon/server"
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/require"
+
+	"github.com/b2aio/typhon/example/handler"
+)
+
+func initServer(t *testing.T) {
+	// Initialize our Server
+	server.Init(&server.Config{
+		Name:        "example",
+		Description: "Example service",
+	})
+
+	// Register an example endpoint
+	server.RegisterEndpoint(handler.Hello)
+	server.RegisterEndpoint(handler.CallHello)
+
+	go server.Run()
+
+	select {
+	case <-server.DefaultServer.NotifyConnected():
```

Comment:
We should probably add a `NotifyConnected()` func at the package level that operates on the `DefaultServer` to streamline this - we can do separately tho.